examples/example_weather_agent.py

```diff
@@ -0,0 +1,268 @@
+            'message': f"Failed to get weather for {city}",
+            'data': None
+        }
+
+def get_state(function_result: Optional[Dict], current_state: Optional[Dict]) -> Dict[str, Any]:
+    """Get the current state of the weather agent.
+    
+    This function maintains the agent's state between function calls,
+    tracking successful and failed requests.
+    
+    Args:
+        function_result: Result of the last executed function
+        current_state: Current agent state
+        
+    Returns:
+        Dict[str, Any]: Updated agent state
+        
+    Example:
+        >>> state = get_state(None, None)
+        >>> print(state)
+        {'requests': 0, 'successes': 0, 'failures': 0}
+    """
+    if current_state is None:
+        return {'requests': 0, 'successes': 0, 'failures': 0}
+        
+    if function_result and function_result.get('status') == 'success':
+        current_state['successes'] += 1
+    elif function_result:
+        current_state['failures'] += 1
+        
+    current_state['requests'] += 1
+    return current_state
+
+def create_weather_agent() -> Agent:
+    """Create and configure a weather agent.
+    
+    This function sets up an agent with a weather worker that can
+    fetch weather information and provide recommendations.
+    
+    Returns:
+        Agent: Configured weather agent
+        
+    Raises:
+        ValueError: If VIRTUALS_API_KEY is not set
+        ValidationError: If worker configuration is invalid
+        APIError: If agent creation fails
+        
+    Example:
+        >>> agent = create_weather_agent()
+        >>> agent.compile()
+    """
+    # Get API key
+    api_key = os.getenv('VIRTUALS_API_KEY')
+    if not api_key:
+        raise ValueError("VIRTUALS_API_KEY not set")
+    
+    logger.debug("Starting weather reporter creation")
+    logger.debug(f"Creating agent with API key: {api_key[:8]}...")
+    
+    # Create weather function
+    weather_fn = Function(
+        fn_name="get_weather",
+        fn_description="Get weather information for a city",
+        executable=get_weather,
+        args=[
+            Argument(
+                name="city",
+                type="string",
+                description="City to get weather for"
+            )
+        ]
+    )
+    
+    # Create worker config
+    worker_config = WorkerConfig(
+        id="weather_worker",
+        worker_description="Provides weather information and recommendations",
+        get_state_fn=get_state,
+        action_space=[weather_fn],
+        instruction="Fetch weather data and provide clothing recommendations",
+        api_key=api_key
+    )
+    
+    # Create agent
+    agent = Agent(
+        api_key=api_key,
+        name="Weather Reporter",
+        agent_description="Reports weather and provides recommendations",
+        agent_goal="Help users prepare for weather conditions",
+        get_agent_state_fn=get_state,
+        workers=[worker_config]
```

Comment:
Oh nice. I didn't realize you set the workers list at initialization instead of adding them later.